### PR TITLE
LOK-2149: Add TopN Nodes GrapQL API

### PR DESCRIPTION
**Pull request by cgorantla**
## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2149

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java [hunk 1]
```diff
@@ -114,4 +116,11 @@ public Mono<Boolean> deleteNode(@GraphQLArgument(name = "id") Long id, @GraphQLE
     public Mono<Boolean> discoveryByNodeIds(List<Long> ids, @GraphQLEnvironment ResolutionEnvironment env) {
         return Mono.just(client.startScanByNodeIds(ids, headerUtil.getAuthHeader(env)));
     }
+
+    @GraphQLQuery
+    public Flux<TopNNode> getTopNNode(@GraphQLEnvironment ResolutionEnvironment env, Integer timeRange, TimeRangeUnit timeRangeUnit) {
```
Comment (freddy61025) [inline rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java:121]: is that function name a bit misleading? It seems it is about reachability.

File: rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java [hunk 2]
```diff
@@ -114,4 +116,11 @@ public Mono<Boolean> deleteNode(@GraphQLArgument(name = "id") Long id, @GraphQLE
...
+        var nodes = client.listNodes(headerUtil.getAuthHeader(env));
+        var topNNodes = nodes.stream().map(nodeDTO -> nodeStatusService.getTopNNode(nodeDTO, timeRange,timeRangeUnit, env)).toList();
```
Comment (freddy61025) [inline rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java:123]: do we want to provide a simple sorting by reachability and max count ?

**Review by freddy61025 — APPROVED** [review]
codewise it is LGTM, just few comment about the requirement.

File: rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java (hunk 1 above)
Comment (cgorantla) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java:121]: It also has average response time. The table corresponds to TopN nodes so it seems appropriate.

File: rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java (hunk 2 above)
Comment (cgorantla) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java:123]: I have a PR coming, will include sorting in that PR

**Review by bouff — APPROVED** [review]

**Pull request merged**